codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamFrameToHttpObjectCodec.java

```diff
@@ -118,8 +133,36 @@ private void encodeLastContent(LastHttpContent last, List<Object> out) {
         }
     }
 
+    /**
+     * Encode from an {@link HttpObject}to an {@link Http2StreamFrame}. This method will
+     * be called for each written message that can be handled by this encoder.
+     *
+     * NOTE: 100-Continue responses that are NOT {@link FullHttpResponse} will be rejected.
+     *
+     * @param ctx           the {@link ChannelHandlerContext} which this handler belongs to
+     * @param obj           the {@link HttpObject} message to encode
+     * @param out           the {@link List} into which the encoded msg should be added
+     *                      needs to do some kind of aggregation
+     * @throws Exception    is thrown if an error occurs
+     */
     @Override
     protected void encode(ChannelHandlerContext ctx, HttpObject obj, List<Object> out) throws Exception {
+        // 100-continue is typically a FullHttpResponse, but the decoded
+        // Http2HeadersFrame should not be marked as endStream=true
+        if (obj instanceof HttpResponse) {
+            final HttpResponse res = (HttpResponse) obj;
+            if (res.status().equals(HttpResponseStatus.CONTINUE)) {
+                if (res instanceof FullHttpResponse) {
+                    final Http2Headers headers = toHttp2Headers(res);
+                    out.add(new DefaultHttp2HeadersFrame(headers, false));
+                    return;
+                } else {
+                    throw new EncoderException(
+                            "100-continue response must be a FullHttpResponse");
```

Comment:
consider using the `HttpResponseStatus.CONTINUE` instead of `100-continue` for consistency.

Reply:
fixed.